Propose Code of Conduct

parthatom opened the pull request:
For every great open source community to function efficiently, a code of conduct is always important.

robert-wagner commented:
Hey @parthatom I'm cool with the contents of this pr but would like feedback from @iamtrask before merging.

That being said, could you make all of the heading in the document headings and all the lists, lists (using markdown formatting)

mari-linhares commented on CODE_OF_CONDUCT.md:
```diff
@@ -0,0 +1,33 @@
+Our Pledge
```
Add title:

# Code of Conduct

mari-linhares replied:
Our Pledge -> ##OurPledge

mari-linhares commented on CODE_OF_CONDUCT.md:
```diff
@@ -0,0 +1,33 @@
+Our Pledge
+In the interest of fostering an open and welcoming environment, we as contributors and maintainers pledge to making participation in our project and our community a harassment-free experience for everyone, regardless of age, body size, disability, ethnicity, sex characteristics, gender identity and expression, level of experience, education, socio-economic status, nationality, personal appearance ...
+
+Our Standards
```
Our Standards -> ##Our Standards

mari-linhares commented on CODE_OF_CONDUCT.md:
```diff
@@ -0,0 +1,33 @@
+Our Standards
+Examples of behavior that contributes to creating a positive environment include:
+
+Using welcoming and inclusive language
```
Make this a list.

mari-linhares commented on CODE_OF_CONDUCT.md:
```diff
@@ -0,0 +1,33 @@
+Showing empathy towards other community members
+Examples of unacceptable behavior by participants include:
+
+The use of sexualized language or imagery and unwelcome sexual attention or advances
```
Make this a list.

mari-linhares commented on CODE_OF_CONDUCT.md:
```diff
@@ -0,0 +1,33 @@
+Public or private harassment
+Publishing others’ private information, such as a physical or electronic address, without explicit permission
+Other conduct which could reasonably be considered inappropriate in a professional setting
+Our Responsibilities
```
Our Responsibilities -> ##Our Responsibilities

mari-linhares commented on CODE_OF_CONDUCT.md:
```diff
@@ -0,0 +1,33 @@
+
+Project maintainers have the right and responsibility to remove, edit, or reject comments, commits, code, wiki edits, issues, and other contributions that are not aligned to this Code of Conduct, or to ban temporarily or permanently any contributor for other behaviors that they deem inappropriate, threatening, offensive, or harmful.
+
+Scope
```
Scope -> ##Scope

mari-linhares commented on CODE_OF_CONDUCT.md:
```diff
@@ -0,0 +1,33 @@
+Scope
+This Code of Conduct applies within all project spaces, and it also applies when an individual is representing the project or its community in public spaces. Examples of representing a project or community include using an official project e-mail address, posting via an official social media account, or acting as an appointed representative at an online or offline event. Representation of a proje ...
+
+Enforcement
```
Enforcement -> ##Enforcement

mari-linhares commented on CODE_OF_CONDUCT.md:
```diff
@@ -0,0 +1,33 @@
+
+Project maintainers who do not follow or enforce the Code of Conduct in good faith may face temporary or permanent repercussions as determined by other members of the project’s leadership.
+
+Attribution
```
Attribution -> ##Attribution

mari-linhares commented:
Oh, I've just seen that there's a PR for creating a code of conduct on syft as well, please consider the request changes for syft's PR as well @parthatom

parthatom commented:
@mari-linhares Thanks for doing the cumbersome task of pointing out every change. I have resolved all of the changes asked and updated the PR. Please go through them.

parthatom commented:
Also linked the code of conduct in the CONTRIBUTING.md Kindly go through it as well!

mari-linhares commented on CODE_OF_CONDUCT.md:
```diff
@@ -0,0 +1,36 @@
+
+##Our Standards
+Examples of behavior that contributes to creating a positive environment include:
+<ul>
```
Oh, you can make a list in markdown with * (as bullet points):

This would look like this:

* Being respectful of differing viewpoints and experiences
* Gracefully accepting constructive criticism
* Focusing on what is best for the community
* Showing empathy towards other community members

But in markdown is actually:

```
* Being respectful of differing viewpoints and experiences
* Gracefully accepting constructive criticism
* Focusing on what is best for the community
* Showing empathy towards other community members
```

parthatom commented:
Github still says changes requested. How do I take care of that?

mari-linhares commented on CONTRIBUTING.md:
```diff
@@ -2,6 +2,10 @@
 
+### Code of Conduct
+
+On <https://github.com/parthatom/Grid/CODE_OF_CONDUCT.md> you can find the code of conduct for an efficient working of this community.
```
You can link files in markdown by doing opening and closing brackets then opening and closing parentheses: [link example ]( ) (`[<link name>](<link url>)`

robert-wagner replied:
correct the link to the openmined grid repo

robert-wagner replied:
wording: You can find our code of conduct at <link>. We ask that all members of the community follow this code in order to promote a welcoming and productive environment.

robert-wagner commented on CONTRIBUTING.md:
```diff
@@ -101,4 +105,4 @@ After each commit TravisCI will check your new code against the formatting guide
 
-If your check fails don't worry you will still be able to make changes and make your code pass the checks.
\ No newline at end of file
+If your check fails don't worry you will still be able to make changes and make your code pass the checks.
```
Why is this line listed as a change?

parthatom replied:
I can't see it when I run git diff. It already existed in the document. Maybe some spaces/enters affected it?

parthatom commented:
@robert-wagner @mari-linhares Thanks again. I have updated the files with the proposed changes in the markdown lists and links. Also changed the wordings and rewrote the link to Code of Conduct. Please go through them.

mari-linhares commented on CODE_OF_CONDUCT.md:
```diff
@@ -0,0 +1,37 @@
+#Code of Conduct
```
Sorry, I've missed this in the last review but can you add a space between the "#" and the title? Otherwise is not displayed correctly.

mari-linhares commented on CONTRIBUTING.md:
```diff
@@ -2,6 +2,11 @@
+### Code of Conduct
+
+
+You can find our code of conduct at [CODE_OF_CONDUCT.md][https://github.com/OpenMined/Grid/CODE_OF_CONDUCT.md]. We ask that all members of the community follow this code in order to promote a welcoming and productive environment.
```
Replace: [CODE_OF_CONDUCT.md][https://github.com/OpenMined/Grid/CODE_OF_CONDUCT.md] to `[CODE_OF_CONDUCT.md](https://github.com/OpenMined/Grid/CODE_OF_CONDUCT.md)`

parthatom commented:
@mari-linhares Aaaand done?

mari-linhares commented:
sgtm, @robert-wagner , @iamtrask any thoughts?

cereallarceny commented:
@mari-linhares @parthatom @robert-wagner This should actually be done inside of a `.github` folder these days: https://help.github.com/en/articles/creating-a-default-community-health-file-for-your-organization

That's standard open-source practice.  Furthermore, I'd actually be in favor of closing this pull request and creating a single `.github` repository whereby all other repos can inherit.  This is explained in the link above.

cereallarceny commented:
I've actually gone ahead and create this repository.  All repositories under the OpenMined ecosystem will inherit from this repo.  @parthatom if you'd like to go to https://github.com/OpenMined/.github and contribute to the code of conduct there, please do so!  😄

Likewise, we will be able to add all other sorts of files which our repos will inherit automatically like contribution guides and issue/pull request templates.  Give this link a good read: https://help.github.com/en/articles/creating-a-default-community-health-file-for-your-organization

Closing this PR in favor of a global approach.
